homeassistant/components/pyscript/__init__.py

```diff
@@ -0,0 +1,293 @@
+                service_desc = {"description": desc, "fields": fields}
+
+            trig_args = {}
+            trig_decorators = {
+                "time_trigger",
+                "state_trigger",
+                "event_trigger",
+                "state_active",
+                "time_active",
+            }
+            for dec in func.get_decorators():
+                dec_name, dec_args = dec[0], dec[1]
+                if dec_name in trig_decorators:
+                    if dec_name not in trig_args:
+                        trig_args[dec_name] = []
+                    if dec_args is not None:
+                        trig_args[dec_name] += dec_args
+                elif dec_name == "service":
+                    if dec_args is not None:
+                        _LOGGER.error(
+                            "%s defined in %s: decorator @service takes no arguments; ignored",
+                            name,
+                            file,
+                        )
+                        continue
+                    _LOGGER.debug(
+                        "registering %s/%s (desc = %s)", DOMAIN, name, service_desc
+                    )
+                    hass.services.async_register(
+                        DOMAIN,
+                        name,
+                        pyscript_service_factory(name, func, global_sym_table),
+                    )
+                    async_set_service_schema(hass, DOMAIN, name, service_desc)
+                    services.add(name)
+                else:
+                    _LOGGER.warning(
+                        "%s defined in %s has unknown decorator @%s",
+                        name,
+                        file,
+                        dec_name,
+                    )
+            for dec_name in trig_decorators:
+                if dec_name in trig_args and len(trig_args[dec_name]) == 0:
+                    trig_args[dec_name] = None
+
+            arg_check = {
+                "state_trigger": {1},
+                "state_active": {1},
+                "event_trigger": {1, 2},
+            }
+            for dec_name, arg_cnt in arg_check.items():
+                if dec_name not in trig_args or trig_args[dec_name] is None:
+                    continue
+                if len(trig_args[dec_name]) not in arg_cnt:
+                    _LOGGER.error(
+                        "%s defined in %s decorator @%s got %d argument%s, expected %s; ignored",
+                        name,
+                        file,
+                        dec_name,
+                        len(trig_args[dec_name]),
+                        "s" if len(trig_args[dec_name]) > 1 else "",
+                        " or ".join(sorted(arg_cnt)),
+                    )
+                    del trig_args[dec_name]
+                if arg_cnt == 1:
+                    trig_args[dec_name] = trig_args[dec_name][0]
+
+            if len(trig_args) > 0:
```

Comment:
Consider using a guard clause to save one more indentation.

```python
if len(trig_args) == 0:
    continue
```

Reply:
Thanks - done in commit below.